### DISCUSSION TITLE
useQuery not doing a deep comparison

### DISCUSSION THREAD

**Discussion by WesleyYue** [TanStack/query#7318] [Q&A]
This is probably something I am doing wrong because I can't figure out a minimal repro with this issue. I have something like:
```
  const { data } = useQuery({
    queryKey: ['repoData'],
    queryFn: async () =>{
      const res = await axios.get('https://api.github.com/...');
      const array = [1, 2, 3];  // Some array I dynamically calculate based on response
      return {id: res.data.id, a: array};
    },
    select: (data) => data.a,
  })
  ```
  
Using `use-what-changed` from https://github.com/simbathesailor/use-what-changed, I observe that `data` returned by useQuery is changing even though they have the same value. I confirm this in that adding `useDeepCompareMemo(() => data, [data])` gets me the behaviour I want. I expected react-query to already be doing a deep compare memo on the selected data internally. Is this correct? If so, what am I doing wrong?

**Comment by TkDodo** [comment]
looking at this code, the component should only re-render if the `array` changes. Code looks fine so without a minimal reproduction, there isn't much I can do, sorry.

**Reply by WesleyYue** [reply]
No problem, thanks. Knowing that there isn't anything obviously wrong here is already helpful!
